src/ILVerify/tests/TestDataLoader.cs

```diff
@@ -0,0 +1,165 @@
+                    {
+                        if (Enum.TryParse(item, out VerifierError expectedError))
+                        {
+                            verificationErros.Add(expectedError);
+                        }
+                    }
+
+                    var newItem = new InvalidILTestCase { MethodHandle = methodHandle };
+
+                    if (expectedErrors.Length > 0)
+                    {
+                        newItem.ExpectedVerifierError = verificationErros;
+                    }
+
+                    return newItem;
+                }
+                return null;
+            });
+            return GetTestMethodsFromDll(methodSelector);
+        }
+
+        private static IEnumerable<Object[]> GetTestMethodsFromDll(Func<string[], MethodDefinitionHandle, TestCase> methodSelector)
+        {
+            List<TestCase[]> retVal = new List<TestCase[]>();
+
+            foreach (var testDllName in GetAllTestDlls())
+            {
+                var testModule = GetModuleForTestAssembly(testDllName);
+
+                foreach (var methodHandle in testModule.MetadataReader.MethodDefinitions)
+                {
+                    var method = (EcmaMethod)testModule.GetMethod(methodHandle);
+                    var methodName = method.ToString();
+
+                    if (!String.IsNullOrEmpty(methodName) && methodName.Contains("_"))
+                    {
+                        var mparams = methodName.Split('_');
+                        var newItem = methodSelector(mparams, methodHandle);
+
+                        if (newItem != null)
+                        {
+                            newItem.MethodName = methodName;
+                            newItem.ModuleName = testDllName;
+
+                            retVal.Add(new TestCase[] { newItem });
+                        }
+                    }
+                }
+            }
+            return retVal;
+        }
+
+        private static IEnumerable<string> GetAllTestDlls()
+        {
+            foreach (var item in System.IO.Directory.GetFiles(TESTASSEMBLYPATH))
+            {
+                if (item.ToLower().EndsWith(".dll"))
+                {
+                    yield return System.IO.Path.GetFileName(item);
+                }
+            }
+        }
+
+        public static EcmaModule GetModuleForTestAssembly(string assemblyName)
+        {
+            var typeSystemContext = new SimpleTypeSystemContext();
+            var coreAssembly = typeof(Object).Assembly;
+            var systemRuntime = Assembly.Load("System.Runtime");
+
+            typeSystemContext.InputFilePaths = new Dictionary<string, string>
+            {
+                { coreAssembly.GetName().Name, coreAssembly.Location },
+                { systemRuntime.GetName().Name, systemRuntime.Location }
+            };
+
+            typeSystemContext.SetSystemModule(typeSystemContext.GetModuleForSimpleName(coreAssembly.GetName().Name));
```

Comment:
Either one will work in current implementation, but `coreAssembly.GetName().Name` is more correct. The system module is meant to be the actual module where the wellknowtypes live, not a forwarder.

>  Is this assumption correct?

Right.